java-frontend/src/main/java/org/sonar/java/matcher/MethodMatchersBuilder.java

```diff
@@ -104,41 +121,52 @@ public Builder constructor() {
 
   @Override
   public Builder name(Predicate<String> namePredicate) {
-    this.namePredicate = or(this.namePredicate, namePredicate);
-    return this;
+    return new MethodMatchersBuilder(typePredicate, or(this.namePredicate, namePredicate), parametersPredicate);
   }
 
   @Override
   public Builder withParameters(String... parametersType) {
-    return withParameters(Arrays.stream(parametersType).
-      <Predicate<Type>>map(parameterType -> (type -> type.is(parameterType)))
+    return withParameters(Arrays.stream(parametersType).<Predicate<Type>>map(parameterType -> (type -> type.is(parameterType)))
       .collect(Collectors.toList()));
   }
 
   @Override
-  public Builder withParameters(Predicate<Type>... parametersType) {
+  @SafeVarargs
+  public final Builder withParameters(Predicate<Type>... parametersType) {
     return withParameters(Arrays.asList(parametersType));
   }
 
   private Builder withParameters(List<Predicate<Type>> parametersType) {
     return withParameters((List<Type> actualTypes) -> exactMatchesParameters(parametersType, actualTypes));
   }
 
+  @Override
+  public MethodMatchers.Builder withoutParameters() {
+    return withParameters(Collections.emptyList());
+  }
+
+  @Override
+  public MethodMatchers.Builder withAnyParameters() {
```

Comment:
Now that we have these two methods, do we still have the need for `startWithParameters`? 
The behavior could be replaced by a call to `withAnyParameters` after `withParameters`?

Reply:
No, currently all parameters pattern are defined by the call of one method, it allows to do a `or`.
if you do: `.withParameters("int").withParameters("long").withAnyParameters()` it raises a new IllegalStateException("Incompatible 'any parameters' constraint added to existing parameters constraint.")